homeassistant/components/rachio/coordinator.py

```diff
@@ -54,3 +66,55 @@ async def _async_update_data(self) -> dict[str, Any]:
         except Timeout as err:
             raise UpdateFailed(f"Could not connect to the Rachio API: {err}") from err
         return {valve[KEY_ID]: valve for valve in data[1][KEY_VALVES]}
+
+
+class RachioScheduleUpdateCoordinator(DataUpdateCoordinator[list[dict[str, Any]]]):
+    """Coordinator for fetching hose timer schedules."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        rachio: Rachio,
+        base_station,
+    ) -> None:
+        """Initialize a Rachio schedule coordinator."""
+        self.hass = hass
+        self.rachio = rachio
+        self.base_station = base_station
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN} schedule update coordinator",
+            update_interval=timedelta(minutes=30),
+        )
+
+    async def _async_update_data(self) -> Any:
+        """Retrieve data for the past week and the next 60 days."""
+        _time_start: datetime = dt_util.now() - timedelta(days=7)
+        _time_end: datetime = dt_util.now() + timedelta(days=60)
```

Comment:
I just picked values here that I thought made sense, but the API allows for up to 100 days of activity, so these can be adjusted if we think more or less is better.